docs/pages/contract-api/evm/functions.mdx

```diff
@@ -0,0 +1,250 @@
+    string params;       // Additional parameters as JSON (optional)
+}
+```
+
+### Value
+
+The function is `payable`. The caller must send at least `getSignatureDeposit()` wei as the signature fee.
+
+### Emits
+
+- `SignatureRequested` event
+
+### Example
+
+```typescript
+import { encodeFunctionData } from 'viem'
+
+const signRequest = {
+  payload: payloadHash, // bytes32
+  path: 'eth', // derivation path
+  keyVersion: 0, // key version
+  algo: '', // algorithm (optional)
+  dest: '', // destination (optional)
+  params: '', // params (optional)
+}
+
+const deposit = await contract.read.getSignatureDeposit()
+
+const hash = await walletClient.writeContract({
+  address: contractAddress,
+  abi: chainSignaturesAbi,
+  functionName: 'sign',
+  args: [signRequest],
+  value: deposit,
+})
+```
+
+---
+
+## respond
+
+Returns signatures from the MPC network. Called by MPC responders after signature generation.
+
+### Signature
+
+```solidity
+function respond(Response[] calldata _responses) external
+```
+
+### Parameters
+
+| Parameter    | Type         | Description                                              |
+| ------------ | ------------ | -------------------------------------------------------- |
+| `_responses` | `Response[]` | Array of responses containing request IDs and signatures |
+
+### Response Struct
+
+```solidity
+struct Response {
+    bytes32 requestId;   // Request identifier
+    Signature signature; // ECDSA signature
+}
+```
+
+### Emits
+
+- `SignatureResponded` event for each response
+
+### Notes
+
+- Only addresses with the MPC_NETWORK role can call this function
+- Array processing is atomic - all responses succeed or all fail
+
+---
+
+## respondError
+
+Reports signature generation errors from the MPC network.
+
+### Signature
+
+```solidity
+function respondError(ErrorResponse[] calldata _errors) external
+```
+
+### Parameters
+
+| Parameter | Type              | Description              |
+| --------- | ----------------- | ------------------------ |
+| `_errors` | `ErrorResponse[]` | Array of error responses |
+
+### ErrorResponse Struct
+
+```solidity
+struct ErrorResponse {
+    bytes32 requestId;     // Request identifier
+    string errorMessage;   // Error description
+}
+```
+
+### Emits
+
+- `SignatureError` event for each error
+
+---
+
+## getSignatureDeposit
+
+Returns the current signature deposit amount required for signing requests.
+
+### Signature
+
+```solidity
+function getSignatureDeposit() external view returns (uint256)
+```
+
+### Returns
+
+| Type      | Description                      |
+| --------- | -------------------------------- |
+| `uint256` | Current signature deposit in wei |
+
+### Example
+
+```typescript
+const deposit = await publicClient.readContract({
+  address: contractAddress,
+  abi: chainSignaturesAbi,
+  functionName: 'getSignatureDeposit',
+})
+
+console.log(`Deposit required: ${formatEther(deposit)} ETH`)
+```
+
+---
+
+## setSignatureDeposit
```

Comment:
same thing here, this and admin management functions are not relevant to developers